estimate-web/src/components/UserStoryCommentList.jsx

```diff
@@ -0,0 +1,93 @@
+import React, { Component } from "react";
+import {
+    Spinner, Persona, PersonaSize, Label
+} from "office-ui-fabric-react";
+import { PropTypes } from "prop-types";
+import { connect } from "react-redux";
+import { requestWorkItemGetComments } from "../actions/devops";
+import workItemShape from "../reducers/models/workItemShape";
+
+import "./UserStoryCommentList.less";
+import timeAgo from "../utils/date";
+
+class UserStoryCommentList extends Component {
+    componentDidMount() {
+        const { workItem, requestComments, commentsFetched } = this.props;
+        if (!commentsFetched) {
+            requestComments(workItem.id);
+        }
+    }
+
+    componentDidUpdate(previousProps) {
```

Comment:
I don't have a concrete solution in hand, but it feels like you favor solutions with these lifecycle methods, where in fact their use should be minimized as they rely on internal state, which should generally be minimized. Something to think about is if it's possible to move the loading to a higher-level component. That said, I'm sure there are patterns out there that could be helpful (and I have no idea about them 😅)

Reply:
I generally agree with the comment. I'd be inclined to change the code if this was a `getDerivedStateFromProps`. But here I'm literally just checking previous with current props. The control doesn't even have internal state (it may have had at the time of review though).